### ISSUE TITLE
Update Dependencies: StyleLint 16.0

### ISSUE THREAD

**Issue by bpfoster** [ronilaukkarinen/gulp-stylelint#10]
StyleLint 16 released recently.  Appears to be some breaking changes affecting gulp-sylelint.

Bunch of these:
`The CommonJS Node.js API is deprecated. See https://stylelint.io/migration-guide/to-16`'

and a failure:
```
TypeError in plugin "gulp-stylelint"
Message:
    formatter is not a function
```

**Comment by rollecode** [comment]
Yes, there are a lot. Instructions here: [Migrating to 16.0.0](https://stylelint.io/migration-guide/to-16/). Simply updating `peerDependencies` won't cut it. I have very little time to investigate, so any help appreciated.

**Comment by bpfoster** [comment]
I took a stab at it in PR #11.  Unit tests are at least passing...

PR #12 includes the same fixes as #11 as well as migrating this project to ESM.

**Comment by smustgrave** [comment]
Hello, just wondering what the new timeline for a release of gulp-stylelint with latest stylelint is to be released ?

**Comment by JacobDB** [comment]
Would also love to see this, seems like every gulp-stylelint fork gets abandoned...

**Comment by rollecode** [comment]
This is not abandoned. We're been using 15 in the projects. On Friday our team finally has the checkup for future upgrades and this is one of them. Will let you guys know after that. Sorry for tjhe delays.

**Comment by JacobDB** [comment]
Good to hear, I don't mean to be pushy about what is a free library, I appreciate the update!

**Comment by rollecode** [comment]
[My notes](https://github.com/developer-stylechain/gulp-stylelint/pull/12#issuecomment-2334254079) from today.

**Comment by charlieartist** [comment]
Hi, @ronilaukkarinen has there been any progress on this? Thanks!

**Comment by rollecode** [comment]
> Hi, [@ronilaukkarinen](https://github.com/ronilaukkarinen) has there been any progress on this? Thanks!

We’re currently running stylelint 15 in our company and dealing with a lot of technical debt. Our next big update for [air-light,](https://github.com/digitoimistodude/air-light) version 10.0.0, will include this upgrade when we switch to parcel.js. It’s a big job with several steps and major breaking changes, so it’ll take time. Also, since we’re moving to [parcel](https://parceljs.org/), we won’t be using gulp-stylelint anymore.

Sorry it’s taking so long, but there’s about a year’s worth of tasks to catch up on, and upgrading stylelint isn’t the highest priority right now.

Feel free to jump in. I posted some comments about this a year ago [here](https://github.com/developer-stylechain/gulp-stylelint/pull/12#issuecomment-2334254079), but I haven’t tested [this one](https://github.com/developer-stylechain/gulp-stylelint/pull/11) yet. You’re welcome to test those PRs and help us out! I’m ready to merge whenever something is finished and stable.

**Comment by rollecode** [comment]
Unfortunately, we've decided to switch to 100% [Parcel](https://parceljs.org/) since it’s faster, and many things related to Gulp seem slow and unmaintained. This means I will no longer contribute to this repo. I'm sorry to disappoint you all. We'll keep our legacy projects at version 15. Contributions are welcome.
